src/utils/importFuncGenerator.ts

```diff
@@ -0,0 +1,137 @@
+import { ASTNode, SourceUnit } from 'solc-typed-ast';
+import { CairoImportFunctionDefinition } from '../ast/cairoNodes';
+import { AST } from '../ast/ast';
+import assert from 'assert';
+import { TranspileFailedError } from '../utils/errors';
+import { warplibImportInfo } from '../warplib/gatherWarplibImports';
+import { Implicits } from './implicits';
+import {
+  createImportFuncFuncDefinition,
+  createImportStructFuncDefinition,
+  ParameterInfo,
+} from './functionGeneration';
+import { getContainingSourceUnit } from './utils';
+
+// Paths
+const STARKWARE_CAIRO_COMMON_ALLOC = 'starkware.cairo.common.alloc';
+const STARKWARE_CAIRO_COMMON_BUILTINS = 'starkware.cairo.common.cairo_builtins';
+const STARKWARE_CAIRO_COMMON_CAIRO_KECCAK = 'starkware.cairo.common.cairo_keccak.keccak';
+const STARKWARE_CAIRO_COMMON_DEFAULT_DICT = 'starkware.cairo.common.default_dict';
+const STARKWARE_CAIRO_COMMON_DICT = 'starkware.cairo.common.dict';
+const STARKWARE_CAIRO_COMMON_DICT_ACCESS = 'starkware.cairo.common.dict_access';
+const STARKWARE_CAIRO_COMMON_MATH = 'starkware.cairo.common.math';
+const STARKWARE_CAIRO_COMMON_MATH_CMP = 'starkware.cairo.common.math_cmp';
+const STARKWARE_CAIRO_COMMON_UINT256 = 'starkware.cairo.common.uint256';
+const STARKWARE_STARKNET_COMMON_SYSCALLS = 'starkware.starknet.common.syscalls';
+
+export function createImportFuncDefinition(
+  path: string,
+  name: string,
+  nodeInSourceUnit: ASTNode,
+  ast: AST,
+  inputs?: ParameterInfo[],
+  outputs?: ParameterInfo[],
+  options?: { acceptsRawDarray?: boolean; acceptsUnpackedStructArray?: boolean },
+) {
+  const sourceUnit = getContainingSourceUnit(nodeInSourceUnit);
+
+  const hasInputs = inputs !== undefined && inputs.length > 0;
+  const hasOutputs = outputs !== undefined && outputs.length > 0;
```

Comment:
The placement of hasIn, out is misleading, they're only used when there is an existing import